code/game/objects/items/storage/pouch.dm

```diff
@@ -35,6 +38,23 @@
 	..()
 	return user.equip_to_appropriate_slot(src)
 
+/obj/item/storage/pouch/update_icon_state()
+	if(!sprite_slots)
+		return
+
+	var/total_weight = 0
+	for(var/obj/item/i in contents)
+		total_weight += i.w_class
```

Comment:
this is only used if the storage doesn't use slots, so it should be in the if statement for that situation, not before. Otherwise its just getting calculated for no reason.

Reply:
Damn. This should've been obvious to me when I wrote it...
`total_weight += i.w_class` now lives where it would get called when w_class is relevant.